### PR TITLE
[SPARK-22302][INFRA] Remove manual backports for subprocess and print explicit message for < Python 2.7

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

Seems there was a mistake - missing import for `subprocess.call`, while refactoring this script a long ago, which should be used for backports of some missing functions in `subprocess`, specifically in < Python 2.7.

Reproduction is:

```
cd dev && python2.6
```

```
>>> from sparktestsupport import shellutils
>>> shellutils.subprocess_check_call("ls")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sparktestsupport/shellutils.py", line 46, in subprocess_check_call
    retcode = call(*popenargs, **kwargs)
NameError: global name 'call' is not defined
```

For Jenkins logs, please see https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/3950/console

Since we dropped the Python 2.6.x support, looks better we remove those workarounds and print out explicit error messages in order to reduce the efforts to find out the root causes for such cases, for example, `https://github.com/apache/spark/pull/19513#issuecomment-337406734`.

## How was this patch tested?

Manually tested:

```
./dev/run-tests
```

```
Python versions prior to 2.7 are not supported.
```

```
./dev/run-tests-jenkins
```

```
Python versions prior to 2.7 are not supported.
```

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @JoshRosen, @holdenk and @shaneknapp, could you take a look and see if makes sense please?

**Comment by shaneknapp** [comment]
yeah, this makes sense.  i don't think we've officially supported 2.6 in a while, esp for tests and i'd be ok w/removing the backports.  this makes for a much clearer exit case.

@rxin  @sameeragarwal

**Review by dongjoon-hyun — COMMENTED** [review]
+1, too.

**Comment by rxin** [comment]
seems fine.

**Comment by holdenk** [comment]
Looks reasonable, my only thing is putting in a forced exited inside of shellutil.py is sort of a random place to the version check.

**Comment by HyukjinKwon** [comment]
@holdenk, hm, should I maybe make add this to `run-tests.py` and `run-tests-jenkins.py`? I wasn't sure where I should put this.

**Comment by shaneknapp** [comment]
yeah, now that i'm looking at this w/a fresh cup of coffee, i agree w/ @holdenk about shellutil.py being a weird place to exit for a python version check.

run-tests.py and run-tests-jenkins.py would definitely make more sense...  we can probably remove the if: statement in shellutil.py altogether (leaving, of course, the subprocess variable names there).

**Comment by HyukjinKwon** [comment]
Sure, thanks. Let me update soon.

File: dev/run-tests [hunk 1]
```diff
@@ -20,4 +20,10 @@
 FWDIR="$(cd "`dirname $0`"/..; pwd)"
 cd "$FWDIR"
 
+PYTHON_VERSION_CHECK=$(python -c 'import sys; print(sys.version_info < (2, 7, 0))')
```
Comment (HyukjinKwon) [inline dev/run-tests]: @holdenk and @shaneknapp, looks I can't just check this within Python scripts.

As we know from the previous issue, `dev/run-tests.py` is not compatible with Python 2.6.x due to dictionary comprehension syntax:

```
python2.6 dev/run-tests.py
```

```
  File "dev/run-tests.py", line 128
    {m: set(m.dependencies).intersection(modules_to_test) for m in modules_to_test}, sort=True)
                                                            ^
SyntaxError: invalid syntax
```

I tried to change this as below:

```
git diff dev/run-tests.py
```

```diff
diff --git a/dev/run-tests.py b/dev/run-tests.py
index 72d148d7ea0..dcec912ae23 100755
--- a/dev/run-tests.py
+++ b/dev/run-tests.py
@@ -27,6 +27,10 @@ import sys
 import subprocess
 from collections import namedtuple

+if sys.version_info < (2, 7):
+    print("[error] Python versions prior to 2.7 are not supported.")
+    sys.exit(-1)
+
 from sparktestsupport import SPARK_HOME, USER_HOME, ERROR_CODES
 from sparktestsupport.shellutils import exit_from_command_with_retcode, run_cmd, rm_r, which
 from sparktestsupport.toposort import toposort_flatten, toposort
```

but it still gives syntax error ahead:

```
python2.6 dev/run-tests.py
```

```
  File "dev/run-tests.py", line 128
    {m: set(m.dependencies).intersection(modules_to_test) for m in modules_to_test}, sort=True)
                                                            ^
SyntaxError: invalid syntax
```

I think we should not force to workaround for Python 2.6 syntax error as it was dropped. So, I just tried to fix both `dev/run-tests` and `dev/run-tests-jenkins` as proposed currently.

File: dev/run-tests-jenkins [hunk 2]
```diff
@@ -26,4 +26,11 @@ FWDIR="$( cd "$( dirname "$0" )/.." && pwd )"
 cd "$FWDIR"
 
 export PATH=/home/anaconda/bin:$PATH
```
Comment (shaneknapp) [inline dev/run-tests-jenkins]: actually, we set this in the jenkins worker config, spark PRB build config, and here...  we can safely remove this line.

**Review by shaneknapp — APPROVED** [review]
other than deleting the extra PATH-setting line, this change looks good to me!

**Comment by HyukjinKwon** [comment]
Thanks for your review @shaneknapp.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**